sources/niLang/src/DataTablePath.cpp

```diff
@@ -122,7 +146,25 @@ tBool __stdcall ParseDataTablePathOp(const achar* aaszPath, tDataTablePathOpPtrC
     }
     // DataTable
     else {
-      aLst.push_back(niNew cDataTablePathOp(eDataTablePathOp_DataTable,it->Chars()));
+      cString name = *it;
+      auto itNext = it + 1;
+      auto beginPredicate = *it == _A("[");
+      auto predicateWillBegin = itNext != vToks.end() && *itNext == _A("[");
```

Comment:
Use "Lookahead" instead of "WillBegin", that's the usual term when writing parsers. `lookaheadIsPredicate`